ui/src/app/common/services/orb.service.ts

```diff
@@ -0,0 +1,212 @@
+import { AgentGroup } from 'app/common/interfaces/orb/agent.group.interface';
+import { Agent } from 'app/common/interfaces/orb/agent.interface';
+import { AgentPolicy } from 'app/common/interfaces/orb/agent.policy.interface';
+import { Dataset } from 'app/common/interfaces/orb/dataset.policy.interface';
+import { Sink } from 'app/common/interfaces/orb/sink.interface';
+import { AgentGroupsService } from 'app/common/services/agents/agent.groups.service';
+import { AgentPoliciesService } from 'app/common/services/agents/agent.policies.service';
+import { AgentsService } from 'app/common/services/agents/agents.service';
+import { DatasetPoliciesService } from 'app/common/services/dataset/dataset.policies.service';
+import { SinksService } from 'app/common/services/sinks/sinks.service';
+import {
+  BehaviorSubject,
+  defer,
+  EMPTY,
+  merge,
+  Observable,
+  Subject,
+  timer,
+} from 'rxjs';
+import {
+  debounceTime,
+  map,
+  retry,
+  shareReplay,
+  switchMap,
+  takeUntil,
+  tap,
+} from 'rxjs/operators';
+
+export const PollControls = {
+  PAUSE: false,
+  RESUME: true,
+};
+
+@Injectable({
+  providedIn: 'root',
+})
+export class OrbService implements OnDestroy {
+  // interval for timer
+  pollInterval = 1000;
+
+  // race timer && forceRefresh until stopPolling
+  private poller$: Observable<number>;
+
+  pollController$: BehaviorSubject<boolean>;
+
+  lastPollUpdate$: Subject<number>;
+
+  // next to stop polling
+  private killPolling: Subject<void>;
+
+  // next to force refresh
+  private forceRefresh: Subject<number>;
+
+  // convenience polled observables
+  // watch all pages available on agents
+  private agents$: Observable<Agent[]>;
+  private groups$: Observable<AgentGroup[]>;
+  private datasets$: Observable<Dataset[]>;
+  private policies$: Observable<AgentPolicy[]>;
+  private sinks$: Observable<Sink[]>;
+
+  private agentsTags$: Observable<string[]>;
+  private groupsTags$: Observable<string[]>;
+  private sinksTags$: Observable<string[]>;
+
+  pausePolling() {
+    this.pollController$.next(PollControls.PAUSE);
+  }
+
+  startPolling() {
+    this.pollController$.next(PollControls.RESUME);
+  }
+
+  refreshNow() {
+    this.forceRefresh.next(1);
+  }
+
+  observe<T>(observable: Observable<T>) {
+    return this.poller$.pipe(
+      switchMap(() =>
+        observable.pipe(
+          tap((_) => {
+            this.lastPollUpdate$.next(Date.now());
+          }),
+        ),
+      ),
+      retry(),
+      shareReplay(1),
+    );
+  }
+
+  constructor(
+    private agent: AgentsService,
+    private dataset: DatasetPoliciesService,
+    private group: AgentGroupsService,
+    private policy: AgentPoliciesService,
+    private sink: SinksService,
+  ) {
+    this.lastPollUpdate$ = new Subject<number>();
+    this.forceRefresh = new Subject<number>();
+    this.killPolling = new Subject<void>();
+
+    this.pollController$ = new BehaviorSubject<boolean>(PollControls.PAUSE);
```

Comment:
setting initial value to RESUME or calling next(RESUME) on this subject enables polling